vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include "pch.h"
+
+#include <Unicode.h>
+#include <Utils.h>
+#include "WinHTTPWebSocketResource.h"
+
+// Windows API
+#include <comutil.h>
+
+using Microsoft::Common::Unicode::Utf8ToUtf16;
+
+using std::function;
+using std::size_t;
+using std::string;
+using std::wstring;
+
+namespace Microsoft::React
+{
+
+WinHTTPWebSocketResource::WinHTTPWebSocketResource(const string& urlString)
+  : m_openSessionFlags{ WINHTTP_FLAG_ASYNC }
+  , m_sessionHandle{ NULL }
+  , m_connectionHandle{ NULL }
+  , m_requestHandle{ NULL }
+  , m_errorStatus{ ERROR_SUCCESS }
+{
+#if 0 // WinHttpCrackUrl does not accept non-HTTP schemes 🙄
+  BSTR bstr = _com_util::ConvertStringToBSTR(urlString.c_str());
+  LPWSTR lpwstr = bstr;
+  SysFreeString(bstr);
```

Comment:
>SysFreeString [](start = 2, length = 13)

is lpwstr still valid after this?

Reply:
Yes.

Reply:
I'm not sure this is correct. You might see the bytes still be there/unchanged but I'm pretty sure the memory is reclaimed. See if you don't get a crash when you turn on AppVerifier for the process. If bstr happened on a page boundary you would AV.

---
In reply to: [393944590](https://github.com/microsoft/react-native-windows/pull/4330#discussion_r393944590) [](ancestors = 393944590)